packages/material-ui/src/Collapse/Collapse.test.js

```diff
@@ -70,279 +53,138 @@ describe('<Collapse />', () => {
+      handleExit(...args);
+      nodeExitHeightStyle = args[0].style.height;
+    };
+    const handleExiting = spy();
+    const handleExited = spy();
 
     before(() => {
-      wrapper = shallow(<Collapse {...props} />);
-      instance = wrapper.dive().instance();
+      wrapper = mount(
+        <Collapse
+          onEnter={handleEnterWrapper}
+          onEntering={handleEnteringWrapper}
+          onEntered={handleEntered}
+          onExit={handleExitWrapper}
+          onExiting={handleExiting}
+          onExited={handleExited}
+        >
+          <div />
+        </Collapse>,
+      );
+      container = wrapper.find('Transition').childAt(0);
+      stub(container.childAt(0).instance(), 'clientHeight').get(() => 666);
+      clock = useFakeTimers();
     });
 
-    describe('handleEnter()', () => {
-      let element;
-
-      before(() => {
-        element = { style: { height: 32 } };
-        instance.handleEnter(element);
-      });
-
-      it('should set element height to 0 initially', () => {
-        assert.strictEqual(element.style.height, '0px', 'should set the height to 0');
-      });
+    after(() => {
+      clock.restore();
     });
 
-    describe('handleEntering()', () => {
-      let heightMock;
-      let element;
-
+    describe('in', () => {
       before(() => {
-        heightMock = 666;
-        element = { style: { height: heightMock, transitionDuration: undefined } };
-        instance.handleEntering(element);
-      });
-
-      it('should set height to the 0', () => {
-        assert.strictEqual(element.style.height, '0px', 'should have 0px height');
+        wrapper.setProps({ in: true });
       });
 
-      it('should call handleEntering', () => {
-        const onEnteringStub = spy();
-        wrapper.setProps({ onEntering: onEnteringStub });
-        instance = wrapper.dive().instance();
-        instance.handleEntering(element);
-
-        assert.strictEqual(onEnteringStub.callCount, 1);
-        assert.strictEqual(onEnteringStub.calledWith(element), true);
-      });
-
-      describe('timeout', () => {
-        let theme;
-        let timeoutMock;
-        let restore;
-
-        before(() => {
-          theme = instance.props.theme;
-          restore = theme.transitions.getAutoHeightDuration;
-          theme.transitions.getAutoHeightDuration = stub().returns('woofCollapseStub');
-          wrapper.setProps({ timeout: 'auto' });
-          instance = wrapper.dive().instance();
+      describe('handleEnter()', () => {
+        it('should set element height to 0 initially', () => {
+          assert.strictEqual(nodeEnterHeightStyle, '0px', 'should set the height to 0');
         });
+      });
 
-        after(() => {
-          theme.transitions.getAutoHeightDuration = restore;
+      describe('handleEntering()', () => {
+        it('should set height to the wrapper height', () => {
+          assert.strictEqual(nodeEnteringHeightStyle, '666px', 'should have 0px height');
```

Comment:
Do we need the assertion description? It's outdated. The expected value should be good enough no?